### PR TITLE
Threshold config poller for Functions

### PR DESCRIPTION
**Pull request by KuphJr**

### CODE REVIEW

File: core/services/ocr2/plugins/functions/config/config_types.proto [hunk 1]
```diff
@@ -9,11 +9,26 @@ enum AggregationMethod {
     AGGREGATION_MEDIAN = 1;
 }
 
+message OracleIDtoKeyShareIndex {
+  uint32 oracle_id = 1;
+  uint32 key_share_index = 2;
+}
+
+message ThresholdReportingPluginConfig {
```
Comment (bolekk) [inline core/services/ocr2/plugins/functions/config/config_types.proto:17]: You should include the one from tdh2, not copy-paste.

File: core/services/relay/evm/functions/threshold_config_poller.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/libocr/gethwrappers2/ocr2aggregator"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+)
+
+type ThresholdConfigPoller interface {
```
Comment (bolekk) [inline core/services/relay/evm/functions/threshold_config_poller.go:18]: Why not FuncionsConfigPoller that internally has a flag target_config={FUNCTIONS, THRESHOLD, S4} ?

File: core/services/relay/evm/functions/threshold_config_poller.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+	destChainLogPoller logpoller.LogPoller
+	addr               common.Address
+}
+
+func configPollerFilterName(addr common.Address) string {
+	return logpoller.FilterName("OCR2ConfigPoller", addr.String())
+}
+
+// NewConfigPoller creates a new ConfigPoller
+func NewThresholdConfigPoller(lggr logger.Logger, destChainPoller logpoller.LogPoller, addr common.Address) (ThresholdConfigPoller, error) {
```
Comment (bolekk) [inline core/services/relay/evm/functions/threshold_config_poller.go:100]: Why not NewFunctionsConfigPoller(target_config, ...) ?

File: core/services/ocr2/plugins/functions/config/config_types.proto (hunk 1 above)
Comment (KuphJr) [reply core/services/ocr2/plugins/functions/config/config_types.proto:17]: According to a quick search, types from a .proto cannot be imported from an external Go package.

Comment (bolekk) [reply core/services/ocr2/plugins/functions/config/config_types.proto:17]: You import from another .proto file, not from a Go package.

File: core/services/relay/evm/functions/threshold_config_poller.go (hunk 3 above)
Comment (KuphJr) [reply core/services/relay/evm/functions/threshold_config_poller.go:100]: We don't want a custom config poller for Functions.  We only want a custom config poller for Threshold & S4, right?

Comment (bolekk) [reply core/services/relay/evm/functions/threshold_config_poller.go:100]: I'm suggesting to have one custom poller to support all 3. If we're already creating a custom one (which obviously depends on Functions config) why not have it support all our plugins?

Comment (KuphJr) [reply core/services/relay/evm/functions/threshold_config_poller.go:100]: I do not think that is possible.  You need to have a separate instance of the config poller for each plugin.  I do not know how to implement a single poller that implements the logic for each plugin.  The ConfigPoller has a particular interface to which we must adhere.  We cannot add an additional flag parameter to any of these functions without violating the interface.

Comment (KuphJr) [reply core/services/relay/evm/functions/threshold_config_poller.go:100]: Let me try more before I confirm this won't work though.

**Pull request closed without merging**